### ISSUE TITLE
Difficulty reproducing Stage1 model performance

### ISSUE THREAD

**Issue by MichaelYu781** [roboterax/video-prediction-policy#19]
Hi,
Thanks for the great work!

I’m currently trying to reproduce the results of your Stage1 model in order to train Stage2. I followed the released code and instructions without any modifications.
Here’s a summary of my attempts and observations:

1. When using your released Stage1 model, I was able to successfully reproduce Stage2, achieving a length of 4.37.
2. However, when I trained Stage1 from scratch and then used the resulting checkpoints (100k and 120k) to train Stage2, the best performance I could get was around 4.0, even after trying multiple Stage2 checkpoints.

I’d like to confirm:
In the Stage1 configuration, what exactly is the value of pretrained_model_path?
Is it the original stabilityai/stable-video-diffusion-img2vid, or a SVD video model fine-tuned on sthv2, openx, and xhand?

Also, if there’s any evaluation metric or validation script available for Stage1, to help verify whether my Stage1 training is on the right track?

My training log is as follows:

<img width="1776" height="1083" alt="Image" src="https://github.com/user-attachments/assets/1a32d8a6-28d6-4b69-b093-9a1b0d84c93b" />

Thank you very much for your time and clarification!

**Comment by Robert-gyj** [comment]
Hi Michael:
During stage 1, we start from original original SVD and co-train SVD on sthv2, openx, xhand and calvin ABC. you can set the calvin sample ratio to 0.1-0.2 and train it for a longer time. 

I remember there is a difference in config between train_svd.yaml and train_svd_calvin.yaml. Sorry for this misalignment. The text `position_encode` in train_svd is true but in train_svd_calvin is False. You can (1)train from original SVD or (2) modify the `position_encode` and continue train from svd-robot.  And make sure this `position_encode` is aligned with   https://github.com/roboterax/video-prediction-policy/blob/75cf37ad0f537627e1a3aa9a04bab70324ec27ff/policy_conf/VPP_Calvinabc_train.yaml#L58C3-L58C24

We mainly validate the stage 1 performance by generate 1-step predictions on validation dataset. You should see something similar in figure 4 of the paper.

Hope it helps!

Yanjiang

**Comment by MichaelYu781** [comment]
Hi Yanjiang,
Thanks for your patient explanation!

I’ve double-checked the following points:

1. Made sure that position_encode is set to False during Stage 1, Stage 2, and evaluation.

2. For the longer training, I started from [stabilityai/stable-video-diffusion-img2vid](https://huggingface.co/stabilityai/stable-video-diffusion-img2vid) and co-trained on sthv2, openx, xhand, and calvin ABC for 180k steps. The Calvin sample ratio was set to 0.2, consistent with the default setting in [train_calvin_svd.yaml](https://github.com/roboterax/video-prediction-policy/blob/main/video_conf/train_calvin_svd.yaml).

However, even after this long time co-training, the Stage 2 performance remains suboptimal (around 4.0, compared to 4.37 when using your released Stage 1 model). Do you have any suggestions on what else I could check or verify in the Stage 1 setup?

The prediction results of stage1 are shown below and appear normal.

https://github.com/user-attachments/assets/52ed610e-283c-4bb6-a455-3fa3a379f4d8

https://github.com/user-attachments/assets/ded36e66-a314-4412-a90a-585df7b554f4
 


Thanks again for your time and detailed guidance!

Best,
Michael
